src/client/mod.rs

```diff
@@ -793,20 +794,21 @@ impl IotHubClient {
     /// }
     /// ```
     pub fn send_d2c_message(&self, mut message: IotMessage) -> Result<()> {
+        let trace_id: u32 = rand::thread_rng().gen();
```

Comment:
not sure about that.
i would expect a uuid or successive numbers.

Reply:
since it is for generic msgs I'm not sure how to define an ID?
regarding random or incremented numbers I don't have a strong opinion. imho it doesn't matter since you have to search for the ID in your logs anyway. and incremeted numbers might overflow...

Reply:
@mlilien switched to incr. values. closed conversation in order to merge. come back if you have still findings.